src/search.cpp

```diff
@@ -413,7 +403,11 @@ int SuggestionSearch::getEstimatedMatches() const
       std::cerr << "Query Parsing failed, Switching to search without index." << std::endl;
     }
   }
-  return mp_internalDb->m_archives[0].findByTitle(m_query.m_query).size();
+
+  if (mp_archive) {
```

Comment:
(I put the comment here, but it is related to several places).
I don't think we need to store a extra pointer to the archive.
`SuggestionSearch` can only (and always) have the internalDb.
Either the internalDb has database and we do a xapian search, either not and we can use
`mp_internalDb->m_archives[0].findByTitle`. (We just need to be sure that internalDb will have correctly set `m_archives` even if there is no database).

Reply:
That makes sense. I have just added a check here 
https://github.com/openzim/libzim/pull/574/commits/b25b0e538caf772acfce548bd9be0942f2e133e3#diff-da923b7afa45cab7add143c4705b54142e46b2afe9a2627d5fa3b3474bdc8aecR108-R113
that handles the case where the database is absent, in both `newSuggestionFormat` case and otherwise..
There are other cases below where the database fails to load(technically not "absent"). Do we need to make our fallback search in those cases as well?

Reply:
I'm not sure of this anymore.
We have to be careful when we add a archive without a associated xapian database.
When we do multisearch, xapian give us the index of the database in which the result is found and we reuse this index to find to corresponding archive.
If we add a archive but not the xapian database, the index may be wrong.
We must be sure that we are not breaking things when we do this. (Or at least document it, by putting a comment telling this should be keep in sync except when we use suggestion mode)

> There are other cases below where the database fails to load(technically not "absent"). Do we need to make our fallback search in those cases as well?

Yes, we can use the "basic" suggestion search as a fallback.

Reply:
@mgautierfr a separate vector like `archives_wo_index` in internal database to store the archives that don't have an index seems like a good option. What do you think?